### DISCUSSION TITLE
openssl 3.x use http method DELETE,TRACE,OPTIONS visit https port not return SSL_R_HTTP_REQUEST error

### DISCUSSION THREAD

**Discussion by xiaodengchao** [openssl/openssl#22739] [Q&A] [answered]
https://github.com/openssl/openssl/blob/72f98c5d5df345f7818d279c0623c95d87175d63/ssl/record/ssl3_record.c#L341
openssl 3.x use http method DELETE,TRACE,OPTIONS visit https port not return SSL_R_HTTP_REQUEST error
I have use curl test https port
curl -ikv -X TRACE    http://${ip}:8443/
curl -ikv -X OPTIONS  http://${ip}:8443/
curl -ikv -X DELETE   http://${ip}:8443/
the fail reason:
ErrorStack: lib: (SSL routines), func: (0), reason: (wrong version number), code: 167772427, line: 359, file: ssl/record/ssl3_record.c
and I have test other 4 method 
curl -ikv -X POST http://${ip}:8443/
curl -ikv -X GET  http://${ip}:8443/
curl -ikv -X PUT  http://${ip}:8443/
curl -ikv -X HEAD http://${ip}:8443/
the fail reason:
ErrorStack: lib: (SSL routines), func: (0), reason: (http request), code: 167772316, line: 349, file: ssl/record/ssl3_record.c

so what about the differences between POST、GET、PUT、 HEAD and TRACE、OPTIONS、DELETE

**Comment by t8m** [comment, marked as answer]
~~I am sorry but this is not a question for OpenSSL but curl.~~

Oh I misunderstood your question. The code in ssl3_record.c just attempts heuristically to detect whether the peer is attempting plain http request to give hint in the error code. It is not meant to be fully comprehensive check. Any other situation where someone connects with other (non-TLS) protocol just usually gives the wrong version number error.
